operator/hack/deploy-aws-storage-secret.sh

```diff
@@ -1,19 +1,23 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -eou pipefail
+set -euo pipefail
 
-BUCKET_NAME=$1
+if [ -z "${1-}" ]; then
+    echo "Provide a bucket name"
+    exit 1
+fi
 
-NAMESPACE=${NAMESPACE:-openshift-logging}
+readonly bucket_name=$1
 
-REGION=${REGION:-$(aws configure get region)}
-ACCESS_KEY_ID=${ACCESS_KEY_ID:-$(aws configure get aws_access_key_id)}
-SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY:-$(aws configure get aws_secret_access_key)}
+readonly namespace=${NAMESPACE:-openshift-logging}
+readonly region=${REGION:-$(aws configure get region)}
+readonly access_key_id=${ACCESS_KEY_ID:-$(aws configure get aws_access_key_id)}
+readonly secret_access_key=${SECRET_ACCESS_KEY:-$(aws configure get aws_secret_access_key)}
```

Comment:
Unfortunately, this does not work properly, because the `readonly` would "swallow" the exit code of the subshell. When using subshells you need to do the `readonly` in a separate call:

```suggestion
access_key_id=${ACCESS_KEY_ID:-$(aws configure get aws_access_key_id)}
readonly access_key_id
secret_access_key=${SECRET_ACCESS_KEY:-$(aws configure get aws_secret_access_key)}
readonly secret_access_key
```